addons/website_form_project/controllers/main.py

```diff
@@ -2,14 +2,13 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo.addons.website.controllers import form
+from odoo.http import request
 
 
 class WebsiteForm(form.WebsiteForm):
-    def insert_record(self, request, model, values, custom, meta=None):
-        if model.model == 'project.task':
-            visitor_sudo = request.env['website.visitor']._get_visitor_from_request()
-            visitor_partner = visitor_sudo.partner_id
-            if visitor_partner:
-                values['partner_id'] = visitor_partner.id
-
-        return super().insert_record(request, model, values, custom, meta=meta)
+    def extract_data(self, model, values):
+        res = super().extract_data(model, values)
+        for field_name, field_value in values.items():
+            if model.model == 'project.task' and field_name == 'email_from':
+                res['record']['partner_id'] = request.env['res.partner'].sudo().search([('email', '=', field_value)]).id
```

Comment:
```suggestion
        if model.model == 'project.task' and 'email_from' in values:
            res['record']['partner_id'] = request.env['res.partner'].sudo().search([('email', '=', values['email_from'])]).id
```
I don't think the loop here over the values dictionary is really needed. You can just do this :innocent: